relayer/src/target.ts

```diff
@@ -0,0 +1,39 @@
+import { ApiPromise } from "@polkadot/api";
+import { AddressOrPair } from "@polkadot/api/submittable/types";
+import { Hash } from "@polkadot/types/interfaces";
+import { Observable } from "@polkadot/types/types";
+import { merge } from "rxjs";
+import { concatMap, map } from "rxjs/operators";
+
+class Target {
+  private api: ApiPromise;
+  private signer: AddressOrPair;
+
+  constructor({ api, signer }: { api: ApiPromise; signer: AddressOrPair }) {
+    this.api = api;
+    this.signer = signer;
+  }
+
+  private sendBlockTx = (block: string): Promise<Hash> => {
```

Comment:
Out of curiosity, why not this?:
```suggestion
  private sendBlockTx(block: string): Promise<Hash> {
```

Reply:
if not an arrow function I need explicitly bind it, otherwise `TypeError: Cannot read property 'api' of undefined`

Reply:
I have never had that issue in typescript and always used the non-arrow version. It wouldn't make any sense to even have that syntax if `this` didn't work :man_shrugging:

Reply:
I can make it like this if this style seems better:
`
  private sendBlockTx(block: string): Promise<Hash> {
    return (
      this.api.tx.feeds
        .put(block)
        // it is required to specify nonce, otherwise transaction within same block will be rejected
        // if nonce is -1 API will do the lookup for the right value
        // https://polkadot.js.org/docs/api/cookbook/tx/#how-do-i-take-the-pending-tx-pool-into-account-in-my-nonce
        .signAndSend(this.signer, { nonce: -1 })
    );
  }

  processBlocks(subscriptions: Observable<string>[]): Observable<void> {
    return merge(...subscriptions).pipe(
      concatMap(this.sendBlockTx.bind(this)),
      map((txHash) => {
        // TODO: clarify if we need to know which tx corresponds to which chain
        console.log(`Transaction sent: ${txHash}`);
      })
    );
  }`

Reply:
Ah, now I see what you mean. In this case it makes sense to have `sendBlockTx` as arrow function (though I still prefer regular one), but `processBlocks()` should be the regular one I think. Not that it changes things in a significant way.

Reply:
I don't mind any style, let's decide on a common one

Reply:
I would prefer non-arrow for sure, but let's merge this as is, it looks good enough for now.